art/attacks/evasion/steal_now_attack_latter.py

```diff
@@ -0,0 +1,747 @@
+#
+# Copyright (C) The Adversarial Robustness Toolbox (ART) Authors 2018
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the "Software"), to deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit
+# persons to whom the Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of the
+# Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE
+# WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+# TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+This module implements the paper: "Steal Now and Attack Later: Evaluating Robustness of Object Detection against Black-box Adversarial Attacks"
+| Paper link: https://arxiv.org/abs/2304.05370
+"""
+
+# pylint: disable=C0302
+
+import logging
+from typing import Optional, Tuple, TYPE_CHECKING
+
+import numpy as np
+import random
+
+from art.attacks.attack import EvasionAttack
+
+if TYPE_CHECKING:
+    # pylint: disable=C0412
+    import torch
+
+logger = logging.getLogger(__name__)
+
+def _bbox_ioa(box1: "torch.tenosr",
+              box2: "torch.tenosr",
+              eps: float = 1e-7) -> "torch.tensor":
+    """ 
+    === NOTE ===
+    This function is copied from YOLOv5 repository (yolov5/utils/metrics.py)
+    === ==== ===
+    Returns the intersection over box2 area given box1, box2. Boxes are x1y1x2y2
+    """
+
+    # Get the coordinates of bounding boxes
+    b1_x1, b1_y1, b1_x2, b1_y2 = box1
+    b2_x1, b2_y1, b2_x2, b2_y2 = box2.T
+
+    # Intersection area
+    inter_area = (np.minimum(b1_x2, b2_x2) - np.maximum(b1_x1, b2_x1)).clip(0) * \
+                 (np.minimum(b1_y2, b2_y2) - np.maximum(b1_y1, b2_y1)).clip(0)
+
+    # box2 area
+    box2_area = (b2_x2 - b2_x1) * (b2_y2 - b2_y1) + eps
+
+    # Intersection over box2 area
+    return inter_area / box2_area
+
+def _drop_block2d(input: "torch.tensor",
+                  p: float,
+                  block_size: int):
+    """
+    === NOTE ===
+    This function is modified from torchvision (torchvision/ops/drop_block.py)
```

Comment:
Could you please include the license text of the file yolov5/utils/metrics.py to this docstring?